runner/views/run_api_view.py

```diff
@@ -652,6 +653,82 @@ def post(self, request):
         return Response(body, status=status.HTTP_202_ACCEPTED)
 
 
+class ArgosMappingViewSet(GenericAPIView):
+    logger = logging.getLogger(__name__)
+
+    serializer_class = ArgosPairingSerializer
+
+    def post(self, request):
+        igo_request_id = request.data.get("igo_request_id", None)
+        argos_slug = request.data.get("argos_slug", None)
+        if igo_request_id and argos_slug:
+            operator_model = Operator.objects.get(slug=argos_slug)
+            operator = OperatorFactory.get_by_model(operator_model, request_id=igo_request_id)
+            # construct_argos_jobs() is sloppily separate from the Operator module
+            from runner.operator.argos_operator.v2_0_0.construct_argos_pair import construct_argos_jobs
+
+            files, cnt_tumors = operator.get_files(operator.request_id)
+            data = operator.build_data_list(files)
+            samples = operator.get_samples_from_data(data)
+            argos_inputs, error_samples = construct_argos_jobs(samples)
+            sample_mapping, filepaths = operator.get_mapping_from_argos_inputs(argos_inputs)
+            if sample_mapping:
+                body = {"details": sample_mapping}
+            else:
+                message = "%s: No samples found." % igo_request_id
+                body = {"details": message}
+        return Response(body, status=status.HTTP_202_ACCEPTED)
+
+
+class ArgosDataClinicalViewSet(GenericAPIView):
+    logger = logging.getLogger(__name__)
+
+    serializer_class = ArgosPairingSerializer
+
+    def post(self, request):
+        igo_request_id = request.data.get("igo_request_id", None)
+        argos_slug = request.data.get("argos_slug", None)
+        if igo_request_id and argos_slug:
```

Comment:
I think this is always True because in serializer both values can't be blank.

Reply:
Removed